src/index.js

```diff
@@ -1,37 +1,114 @@
 import './style.css';
 
-const taskContainer = document.getElementById('tasks');
-
-const toDoArr = [
-  {
-    desc: 'I will clean my room',
-    bool: 'true',
-    index: 0,
-  },
-  {
-    desc: 'I have to read a book',
-    bool: 'true',
-    index: 1,
-  },
-  {
-    desc: 'Play fifa with Camavinga',
-    bool: 'false',
-    index: 2,
-  },
-];
-
-const getTasks = () => {
-  toDoArr.forEach((activity, index) => {
-    taskContainer.innerHTML += `
-      <li class="activity">
-        <div class="checkBoxex">
-          <input type="checkbox" name="action1" id="${index}"/>
-          <label for="${index}">${activity.desc}</label><br>
+const form = document.getElementById('todoform');
+const todoInput = document.getElementById('newtodo');
+const toDoListEl = document.getElementById('todos-list');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let EditTodoId = -1;
+
+const renderTodos = () => {
+  toDoListEl.innerHTML = '';
+
+  todos.forEach((todo, index) => {
+    toDoListEl.innerHTML += `
+        <div class="todo" id="${index}">
+            <i 
+              class="bi ${todo.checked ? 'bi-check-circle-fill' : 'bi-circle'}"
+              style="color : ${todo.color}"
+              data-action="check"
+              ></i>
+            <p class="" data-action="check">${todo.value}</p>
+            <i class="bi bi-pencil-square" data-action="edit"></i>
+            <i class="bi bi-trash" data-action="delete"></i>
         </div>
-        <i class="fa-solid fa-ellipsis-vertical fs-3 "></i>
-      </li>
-    `;
+        `;
   });
 };
 
-getTasks();
\ No newline at end of file
+renderTodos();
+
+// eslint-disable-next-line consistent-return
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+
+  const isEmpty = todoValue === '';
+
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+
+  if (isEmpty) {
+    return '';
+  } if (isDuplicate) {
+    return '';
+  }
+  if (EditTodoId >= 0) {
+    todos = todos.map((todo, index) => ({
+      ...todo,
+      value: index === EditTodoId ? todoValue : todo.value,
+    }));
+    EditTodoId = -1;
+  } else {
+    todos.push({
+      value: todoValue,
+      checked: false,
+      color: 'black',
+    });
```

Comment:
- [ ] Could you kindly update the "checked" key to "completed," as outlined in the project requirements? Your attention to this detail would be much appreciated! 🙌

   ![image](https://github.com/Chimoneg27/To-Do-List/assets/107952724/307eb375-44ab-48a5-b001-7454d32f79a3)